pkg/asset/installconfig/gcp/validation.go

```diff
@@ -15,11 +15,28 @@ import (
 func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 
+	allErrs = append(allErrs, validateProject(client, ic, field.NewPath("platform").Child("gcp"))...)
 	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
 
 	return allErrs.ToAggregate()
 }
 
+func validateProject(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.ProjectID != "" {
+		projects, err := client.GetProjects(context.TODO())
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("project"), ic.GCP.ProjectID, err.Error()))
```

Comment:
so this should be an InternalError, because it didn't have anything to do with the project field technically.